plugins/uefi/efi/fwupdate.c

```diff
@@ -427,8 +427,9 @@ fwup_delete_boot_entry(VOID)
 		 * check with EFI_LOAD_OPTION description
 		 */
 		EFI_LOAD_OPTION *load_op = (EFI_LOAD_OPTION *) info_ptr;
-		if (_StrHasPrefix(load_op->description, L"Linux Firmware Updater") ||
-		    _StrHasPrefix(load_op->description, L"Linux-Firmware-Updater")) {
+
+		if (_StrHasPrefix(&load_op->description, L"Linux Firmware Updater") ||
+		    _StrHasPrefix(&load_op->description, L"Linux-Firmware-Updater")) {
```

Comment:
Are you sure? isn't this defined as `CHAR16 description[]`?

Reply:
I kind of think it doesn't matter - I think he should just squash 
1c67712 and 
c54cc1d together anyway rather than ping pong the patches

Reply:
I had tested couple Dell machines and the Intel UEFI develop platform "Rainbowpass", with these patches can successfully do the firmware update and delete the Linux Firmware Updater bootentry. Without "&",  got hang, I think it might be from the way to get the variable.

Reply:
../plugins/uefi/efi/fwupdate.c:430:21: warning: passing argument 1 of ‘_StrHasPrefix’ from incompatible pointer type [-Wincompatible-pointer-types]
  430 |   if (_StrHasPrefix(&load_op->description, L"Linux Firmware Updater") ||
      |                     ^~~~~~~~~~~~~~~~~~~~~
      |                     |
      |                     CHAR16 ** {aka short unsigned int **}
../plugins/uefi/efi/fwupdate.c:370:32: note: expected ‘const CHAR16 *’ {aka ‘const short unsigned int *’} but argument is of type ‘CHAR16 **’ {aka ‘short unsigned int **’}

Reply:
Thanks, looks like I used the older version of fwupd. Will check again the latest version fwupd.